hapi-fhir-spring-boot/hapi-fhir-spring-boot-autoconfigure/src/main/java/ca/uhn/fhir/spring/boot/autoconfigure/FhirAutoConfiguration.java

```diff
@@ -88,6 +89,8 @@
 @Configuration
 @AutoConfigureAfter({DataSourceAutoConfiguration.class, HibernateJpaAutoConfiguration.class})
 @EnableConfigurationProperties(FhirProperties.class)
+@EnableWebSecurity
+@Import({ KeycloakSecurityConfig.class })
```

Comment:
- It works OK when tested locally
- However, does this mean that we will have to rewrite this when we add the authorization piece? I believe that for us to achieve authorization using the SecurityConfig supposed by bootstrap then we'll have to implement ACL annotations on the Controller routes and not centrally from a single class

Reply:
@rehammuzzamil That's correct, we will be able to use `AntMatchers` with Oauth2.0 scope definitions that define the ResourceType scope allowed for the user eg. `antMatchers("/Patient").hasAuthority("SCOPE_NURSE")`. However, I think we will initially implement restrictions based on your "Location" eg. A practitioner in District 1 should not access data in District 2. Is this going to be easy to implement and maintain with Spring security?

Reply:
@ekigamba We need to explore spring security more if there is a need to implement custom restrictions that depend on some other entity like Location etc. Though, there is also one more option in which code level change is not required. Configure Authorization policies through keycloak UI (need to explore it in detail).

Reply:
@rehammuzzamil We can use this for the time being then. What was the reason for not using the Interceptors? I assume that these would have provided more customisation from a central class

Reply:
@ekigamba During the implementation, I figured out that we no longer need the use of Interceptor, as `@KeycloakConfiguration` does what we exactly need. We need to ensure the validation of the token (perform authentication) could be handled by a Configuration class. Please let me know if you have any other thoughts on this. 

I believe if we manage authorization on a keycloak console, it will be managed on a central location with minimal code-level changes. We still have to finalize the next steps for Authorization after further investigation.